src/inputs/NumberField.tsx

```diff
@@ -71,12 +73,18 @@ export function NumberField(props: NumberFieldProps) {
   // If formatOptions isn't memo'd, a useEffect in useNumberStateField will cause jank,
   // see: https://github.com/adobe/react-spectrum/issues/1893.
   const formatOptions: Intl.NumberFormatOptions | undefined = useMemo(() => {
+    if (props.numberFormatOptions !== undefined) {
+      return props.numberFormatOptions;
+    }
+
     return type === "percent"
       ? { style: "percent", signDisplay, ...fractionFormatOptions }
       : type === "basisPoints"
       ? { style: "percent", minimumFractionDigits: 2, signDisplay }
       : type === "cents"
       ? { style: "currency", currency: "USD", minimumFractionDigits: 2, signDisplay }
+      : type === "dollars"
+      ? { style: "currency", currency: "USD", minimumFractionDigits: numFractionDigits ?? 0, signDisplay }
```

Comment:
I like fewer types :).  Maybe I'll change the default here to 2, then at least it will default to cents as you suggest and you can force to dollars only by pasing in numFractionDigits=0

Reply:
thanks for the feedback!

Reply:
eh, but then i need to also override numFractionDigits to 2, if type==dollars... 

maybe as-is is better, the numFractionDigits works as it does everywhere else, and if you want cents, you need to specify numFractionDigits=2 anyway to not avoid truncation...